NetTopologySuite.IO.PostGis/PostGisWriter.cs

```diff
@@ -43,43 +44,20 @@ public PostGisWriter(ByteOrder encodingType)
         /// <inheritdoc cref="IGeometryWriter{TSink}.Write(IGeometry)"/>
         public byte[] Write(IGeometry geometry)
         {
-            return Write(geometry, HandleOrdinates);
-        }
-
-        /// <inheritdoc cref="IGeometryWriter{TSink}.Write(IGeometry, Stream)"/>
-        public void Write(IGeometry geometry, Stream stream)
-        {
-            Write(geometry, HandleOrdinates, stream);
-        }
-
-        /// <summary>
-        /// Writes a binary encoded PostGIS of the given <paramref name="geometry"/> to to an array of bytes.
-        /// </summary>
-        /// <param name="geometry">The geometry</param>
-        /// <param name="ordinates">The ordinates of each geometry's coordinate. <see cref="Ordinates.XY"/> area always written.</param>
-        /// <returns>An array of bytes.</returns>
-        private byte[] Write(IGeometry geometry, Ordinates ordinates)
-        {
-            int coordinateSpace = 8 * OrdinatesUtility.OrdinatesToDimension(ordinates);
+            var maxCoords = HandleOrdinates == Ordinates.None ? Ordinates.XYZM : HandleOrdinates;
+            int coordinateSpace = 8 * OrdinatesUtility.OrdinatesToDimension(maxCoords);
             byte[] bytes = GetBytes(geometry, coordinateSpace);
-            Write(geometry, ordinates, new MemoryStream(bytes));
+            Write(geometry, new MemoryStream(bytes));
```

Comment:
Wasn't there talk recently of a forthcoming poolable memory stream? Does that sound familiar to anyone else?

Reply:
Ah, that does looks like what I was remembering (albeit not quite so recent anymore). But your explanation makes sense, thanks!

Reply:
I personally don't care much about the particular `Write()` overload which creates the MemoryStream, since when actual I/O is performed against PostgreSQL an actual network stream is used instead...